gammapy/maps/wcs.py

```diff
@@ -594,6 +596,30 @@ def upsample(self, factor):
     def to_slice(self, slices):
         raise NotImplementedError
 
+    def solid_angle(self):
+        """ Returns solid angle array (in `sr`) as `~astropy.units.Quantity`
+        """
+        # TODO: Improve by exposing a mode 'edge' for get_coord
+        # Note that edge is applied only to spatial coordinates in the following call
+        pix = self._get_pix_coords(mode='edge')
+        # Note also that pix_to_coord is already called in _get_pix_coords. This should be made more efficient.
+        coords = self.pix_to_coord(pix)
+        lon=coords[0] * np.pi/180.
```

Comment:
We shouldn't litter our code with `np.pi/180` a lot.
Ideally `coords` would have methods / properties that allow you to do something like `lon = coords.lon.to('rad')`.
Does it or is this easy to add?

Reply:
Well `coords` is simply a tuple of arrays, so one needs to create a SkyCoord. But then you have to know which coordinate system you are working with to do either `coords.l`or `coords.ra`.
The function `~gammapy.maps.geom.skycoord_to_lonlat`does it already but it does not return a `Quantity`or an `Angle` but directly an array in degrees. 
A possibility then is to add an option to this function to return radians instead of degrees. 

It all boils down to a better integration of units into `gammapy.maps`.

Reply:
OK, then let's defer this point to another day / PR.
As you know, I'm +1 to use MapCoords objects more instead of raw tuples.